torch/_higher_order_ops/triton_kernel_wrap.py

```diff
@@ -531,6 +539,15 @@ class TritonKernelWrapperFunctional(HigherOrderOperator):
     def __init__(self) -> None:
         super().__init__("triton_kernel_wrapper_functional")
 
+    def __call__(self, kernel_idx, constant_args_idx, grid, kwargs, tensors_to_clone):
+        return self.call_dispatch(
```

Comment:
Why do we need people to call `call_dispatch`? It felt cleaner when the instructions were: "subclass HigherOrderOp and call super().__call__".

You can add a metaclass for HigherOrderOp that actually checks the user is providing an implementation of `__call__` instead.

Reply:
> Why do we need people to call call_dispatch?

We want it to error out earlier than when they construct an instance. Maybe that's not necessary as they'll construct an instance anyway.

>  add a metaclass for HigherOrderOp that actually checks the user is providing an implementation of __call__ instead.

yeah, we could switch to this.